core/src/main/java/io/spine/core/Events.java

```diff
@@ -183,16 +201,78 @@ public static EventId checkValid(EventId id) {
 
     /**
      * Obtains the {@link TenantId} from the given {@link Event}.
+     * 
+     * <p>The {@code TenantId} is retrieved by traversing the passed {@code Event}s context. It is
+     * stored in the initial {@link CommandContext} and can be retrieved from the events origin 
+     * command or rejection context. 
      */
     @Internal
     public static TenantId getTenantId(Event event) {
-        final TenantId result = event.getContext()
-                                     .getCommandContext()
-                                     .getActorContext()
-                                     .getTenantId();
+        final Optional<CommandContext> commandContext = getCommandContext(event);
```

Comment:
Let's append 
```java
checkNotNull(event);
```
as the first line.

`EventsShould` should have a "nullability check" test case (see other Spine tests for a sample).